app/src/main/java/com/linecorp/linesdktest/MenuFragment.java

```diff
@@ -105,9 +128,27 @@ private int convertRadioButtonIdToSaveSlotId(int radioButtonId) {
         }
     }
 
+    @Nullable
+    private Locale getSelectedUILocale() {
+        try {
+            return LocaleUtils.toLocale((String) uiLocaleSpinner.getSelectedItem());
```

Comment:
when its index is 0, LocaleUtils won't find any local, so it will go to catch expression.
But it is not a good way to return null, since index 0 will be a common case.

what about using system current locale when user does not select any item here?
https://stackoverflow.com/a/49256800/1265915

Also, in SDK implementation, since server side already support this flag, if user does not pass any specific locale setting, we could by default use device current locale for them too at `BrowserAuthenticationApi.java ` line 166.

Reply:
I implemented it this way on purpose.

(1) for `LineAuthenticationParams` class:
value `null` for param `uiLocale` means `to use the default locale of client device`
There is no necessary to get Android current locale setting and send it to Server as `ui_locales` param, Web Browser will use its locale setting, and send it as `accept-language` HTTP Request Header to LINE Login Server.
This is the "normal" way how HTTP Server and Client Device negotiate which language to use.

(2) for Test App:
I added a `"[Use Default]"` item to `uiLocaleSpinner`, and its index is 0.
so, if when index is 0, `getSelectedUILocale` will return a `null` value, which means "Use Default"
Without this special selection item, it will be a little difficult to unselect language setting after user select one.

Reply:
For (1), thanks for the explanation.
As for (2), is it better to check whether `uiLocaleSpinner.getSelectedItemPosition()` is 0, if so, return null directly, instead of using `try{} catch {}` for index 0 case?

Reply:
I extracted that special selection item `"[Use Default]"` to a String constant,
and check whether `uiLocaleSpinner.getSelectedItem()` equals to this constant,
so, no `try/catch` now,
is that OK?